### PR TITLE
Fixed captions in Video component

### PR DESCRIPTION
**Pull request by jcfilben**
#### What does this PR do?
This PR fixes the video captions so they will display correctly when the caption button is clicked. The user can also specify multiple caption tracks for different languages and pass in a label to the track element to name different caption tracks for the video.

#### What testing has been done on this PR?
Added captions to the control below story. Also tested by adding multiple captions tracks to one video and adding labels to the track element.

#### How should this be manually tested?
^see above

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?
https://github.com/grommet/grommet/issues/6021
#### Do the grommet docs need to be updated?
No
#### Should this PR be mentioned in the release notes?
Yes
#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

File: src/js/components/Video/Video.js [hunk 1]
```diff
@@ -161,9 +161,10 @@ const Video = forwardRef(
         const { textTracks } = video;
         if (textTracks.length > 0) {
           if (textTracks.length === 1) {
-            const active = textTracks[0].mode === 'showing';
+            const track = textTracks[0];
+            const active = track.mode === 'showing';
             if (!captions || !captions[0] || captions[0].active !== active) {
-              setCaptions([{ active }]);
+              setCaptions([{ label: track.label, active }]);
```
Comment (britt6612) [inline src/js/components/Video/Video.js:167]: maybe add a comment as to even if there is one `track` we still want to grab that  `label` if there  is one

**Review by britt6612 — APPROVED** [review]
Besides adding a comment looks good!  
Just took us a while to understand logic so adding a comment or  2 can help in the future!

File: src/js/components/Video/Video.js (hunk 1 above)
Comment (jcfilben) [reply src/js/components/Video/Video.js:170]: Added a comment here as well as adding comments a few other places in the code.

File: src/js/components/Video/Video.js [hunk 2]
```diff
@@ -278,13 +285,21 @@ const Video = forwardRef(
...
           <Icons.ClosedCaption color={iconColor} />
         ),
         label: caption.label,
         active: caption.active,
-        onClick: () => showCaptions(caption.active ? -1 : 0),
+        a11yTitle: caption.label || 'video.captions',
+        onClick: () => {
+          showCaptions(caption.active ? -1 : index);
+          // set the currently selected captions to active
+          captions[index].active = !captions[index].active;
```
Comment (taysea) [inline src/js/components/Video/Video.js:298]: Is there a reason we aren't using the `setCaptions` function here?

Comment (jcfilben) [reply src/js/components/Video/Video.js]: Updated this section to use `setCaptions` in latest commit

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**